### ISSUE TITLE
[Common] Linear solvers are not found

### ISSUE THREAD

**Issue by rubenzorrilla** [KratosMultiphysics/GiDInterface#512]
Probably because of recent changes in the linear solvers construction, some of the GiD GUI solvers have been stopped working, mainly the ones in the ExternalSolversApplication.

It is quite urgent to fix this, specially with the next release getting closer. Besides, I think is worth it to take the chance to add the new linear solvers which are not yet available in the GUI.

@RiccardoRossi @philbucher @loumalouomega as far as I know, you've done the majority of linear solver related developments, so even though I'll take care of this, I will probably need some help from you.

**Comment by philbucher** [comment]
Can you please post the error-msg you are getting along with the files (ProjectParameters)?

Otherwise it is hard to say whats going on

**Comment by philbucher** [comment]
I just tried with a simple example, selecting the SuperLU and it worked fine:
~~~json
"linear_solver_settings"             : {
    "solver_type" : "ExternalSolversApplication.SuperLUSolver",
    "scaling"     : false
},
~~~

This is exactly what one is supposed to get

**Comment by RiccardoRossi** [comment]
@philbucher did u try from the gid gui?

**Comment by philbucher** [comment]
@RiccardoRossi sure, I mean, how else would I test the gid-gui ;)

I use it still a lot and to my knowledge there are no problems currently

**Comment by rubenzorrilla** [comment]
> I just tried with a simple example, selecting the SuperLU and it worked fine:
> 
> ```json
> "linear_solver_settings"             : {
>     "solver_type" : "ExternalSolversApplication.SuperLUSolver",
>     "scaling"     : false
> },
> ```
> 
> This is exactly what one is supposed to get

This didn't work for me yesterday... Let me check again.

**Comment by philbucher** [comment]
The necessary changes in Kratos were merged 4 days ago: https://github.com/KratosMultiphysics/Kratos/pull/3635

**Comment by jginternational** [comment]
I think it's working now

**Comment by jginternational** [comment]
Using the last compilation of the kratos master branch:

```
Traceback (most recent call last):
  File "MainKratos.py", line 36, in <module>
    simulation = ConvectionDiffusionAnalysisWithFlush(model,parameters)
  File "MainKratos.py", line 16, in __init__
    super(ConvectionDiffusionAnalysisWithFlush,self).__init__(model,project_parameters)
  File "E:\PROYECTOS\KratosGiD\kratos.gid\exec\kratos\applications\convection_diffusion_application\python_scripts\convection_diffusion_analysis.py", line 38, in __init__
    super(ConvectionDiffusionAnalysis, self).__init__(model, project_parameters)
  File "E:\PROYECTOS\KratosGiD\kratos.gid\exec\kratos\kratos\python_scripts\analysis_stage.py", line 41, in __init__
    self._GetSolver().AddVariables() # this creates the solver and adds the variables
  File "E:\PROYECTOS\KratosGiD\kratos.gid\exec\kratos\kratos\python_scripts\analysis_stage.py", line 192, in _GetSolver
    self._solver = self._CreateSolver()
  File "E:\PROYECTOS\KratosGiD\kratos.gid\exec\kratos\applications\convection_diffusion_application\python_scripts\convection_diffusion_analysis.py", line 50, in _CreateSolver
    return solver_wrapper.CreateSolverByParameters(self.model, self.project_parameters["solver_settings"],self.project_parameters["problem_data"]["parallel_type"].GetString())
  File "E:\PROYECTOS\KratosGiD\kratos.gid\exec\kratos\applications\convection_diffusion_application\python_scripts\python_solvers_wrapper_convection_diffusion.py", line 45, in CreateSolverByParameters
    solver = solver_module.CreateSolver(model, solver_settings)
  File "E:\PROYECTOS\KratosGiD\kratos.gid\exec\kratos\applications\convection_diffusion_application\python_scripts\coupled_fluid_thermal_solver.py", line 15, in CreateSolver
    return CoupledFluidThermalSolver(main_model_part, custom_settings)
  File "E:\PROYECTOS\KratosGiD\kratos.gid\exec\kratos\applications\convection_diffusion_application\python_scripts\coupled_fluid_thermal_solver.py", line 55, in __init__
    self.fluid_solver = python_solvers_wrapper_fluid.CreateSolverByParameters(self.model, self.settings["fluid_solver_settings"],"OpenMP")
  File "E:\PROYECTOS\KratosGiD\kratos.gid\exec\kratos\applications\FluidDynamicsApplication\python_scripts\python_solvers_wrapper_fluid.py", line 57, in CreateSolverByParameters
    solver = solver_module.CreateSolver(model, solver_settings)
  File "E:\PROYECTOS\KratosGiD\kratos.gid\exec\kratos\applications\FluidDynamicsApplication\python_scripts\navier_stokes_solver_vmsmonolithic.py", line 115, in CreateSolver
    return NavierStokesSolverMonolithic(model, custom_settings)
  File "E:\PROYECTOS\KratosGiD\kratos.gid\exec\kratos\applications\FluidDynamicsApplication\python_scripts\navier_stokes_solver_vmsmonolithic.py", line 235, in __init__
    self.linear_solver = linear_solver_factory.ConstructSolver(self.settings["linear_solver_settings"])
  File "E:\PROYECTOS\KratosGiD\kratos.gid\exec\kratos\kratos\python_scripts\linear_solver_factory.py", line 37, in ConstructSolver
    return new_linear_solver_factory.ConstructSolver(configuration)
  File "E:\PROYECTOS\KratosGiD\kratos.gid\exec\kratos\kratos\python_scripts\new_linear_solver_factory.py", line 16, in ConstructSolver
    return KratosMultiphysics.LinearSolverFactory().Create(configuration)
RuntimeError: Error: Trying to construct a Linear solver with solver_type:
"ExternalSolversApplication.SuperLUSolver" which does not exist.
The list of available options (for currently loaded applications) is:
Kratos components
    AMGCL
    AMGCLSolver
    AMGCL_NS_Solver
    BICGSTABSolver
    CGSolver
    DeflatedCGSolver
    GMRESSolver
    ScalingSolver
    SkylineLUFactorizationSolver
    SparseLUSolver
    SparseQRSolver
    SuperLUIterativeSolver
    SuperLUSolver
    Super_LU
    TFQMRSolver
    eigen_sparse_lu


in kratos/includes/linear_solver_factory.h:104:LinearSolverFactory<class UblasSpace<double,class boost::numeric::ublas::compressed_matrix<...>,class boost::numeric::Vector >,class UblasSpace<double,class boost::numeric::ublas::matrix<double,...>,class boost::numeric::Vector > >::Create
```

**Comment by philbucher** [comment]
Now it should be globally fixed, can you please pull kratos master and try again?

**Comment by philbucher** [comment]
Guys in the future PLEASE attach the cases you have problems with, otherwise I have no chance of helping if I cannot reproduce the errors

**Comment by rubenzorrilla** [comment]
Hi @philbucher it is working again with the short names (without the `ExternalSolversApplication.`). Maybe we can take the chance to discuss if one of the linear solvers missing in the GUI but present in the list above can be also included.

**Comment by rubenzorrilla** [comment]
> Guys in the future PLEASE attach the cases you have problems with, otherwise I have no chance of helping if I cannot reproduce the errors

**Comment by rubenzorrilla** [comment]
(I accidentally closed it)

**Comment by philbucher** [comment]
> Hi @philbucher it is working again with the short names (without the `ExternalSolversApplication.`). Maybe we can take the chance to discuss if one of the linear solvers missing in the GUI but present in the list above can be also included.

Yes this is working, but now with the current master you will get deprecation-warnings, saying that the name of the Application should be there (as you initially had it)

I discussed with @jcotela abt https://github.com/KratosMultiphysics/Kratos/pull/3966 , bcs this will change the name of the linear-solvers to be mpi/non-mpi- and style-guide-consistent

This is currently waiting for review of @KratosMultiphysics/technical-committee , if this will be merged before Friday then we can use the final names, otherwise we can use the current status

**Comment by rubenzorrilla** [comment]
> > Hi @philbucher it is working again with the short names (without the `ExternalSolversApplication.`). Maybe we can take the chance to discuss if one of the linear solvers missing in the GUI but present in the list above can be also included.
> 
> Yes this is working, but now with the current master you will get deprecation-warnings, saying that the name of the Application should be there (as you initially had it)
> 
> I discussed with @jcotela abt [KratosMultiphysics/Kratos#3966](https://github.com/KratosMultiphysics/Kratos/pull/3966) , bcs this will change the name of the linear-solvers to be mpi/non-mpi- and style-guide-consistent
> 
> This is currently waiting for review of @KratosMultiphysics/technical-committee , if this will be merged before Friday then we can use the final names, otherwise we can use the current status

@ipouplana just FYI.

**Comment by jginternational** [comment]
@rubenzorrilla will you adapt the problemtype to KratosMultiphysics/Kratos#3966 ?

**Comment by philbucher** [comment]
> @rubenzorrilla will you adapt the problemtype to [KratosMultiphysics/Kratos#3966](https://github.com/KratosMultiphysics/Kratos/pull/3966) ?

I will help you, haven't forgotten abt it :)

**Comment by philbucher** [comment]
this is solved now isn't it?

**Comment by rubenzorrilla** [comment]
I think we solved it in #520 . Closing!
